backend/headless/backend.c

```diff
@@ -84,6 +87,44 @@ static void handle_display_destroy(struct wl_listener *listener, void *data) {
 	backend_destroy(&backend->backend);
 }
 
+static bool check_rgba8_ext(void) {
+	const char *exts_str = (const char *)glGetString(GL_EXTENSIONS);
+	if (exts_str == NULL) {
+		wlr_log(WLR_ERROR, "glGetString(GL_EXTENSIONS) failed");
+		return false;
+	}
+	return strstr(exts_str, "GL_OES_rgb8_rgba8") != NULL;
+}
+
+static bool backend_init(struct wlr_headless_backend *backend,
+		struct wl_display *display, struct wlr_renderer *renderer) {
+	wlr_backend_init(&backend->backend, &backend_impl);
+	backend->display = display;
+	wl_list_init(&backend->outputs);
+	wl_list_init(&backend->input_devices);
+
+	backend->renderer = renderer;
+	backend->egl = wlr_gles2_renderer_get_egl(renderer);
+
+	if (!wlr_egl_make_current(backend->egl, EGL_NO_SURFACE, NULL)) {
+		wlr_log(WLR_ERROR, "wlr_egl_make_current failed");
+		return false;
+	}
+
+	if (check_rgba8_ext()) {
+		backend->internal_format = GL_RGBA8_OES;
+	} else {
+		wlr_log(WLR_INFO, "GL_OES_rgb8_rgba8 not supported, "
+			"falling back to GL_RGBA4 internal format");
```

Comment:
WLR_INFO is user-visible, we should give them more information about what this means